File: test-projects/php-sample/src/Logger.php
<?php

namespace CodeGraph\Sample;

/**
 * Log level enumeration
 */
enum LogLevel: int
{
    case DEBUG = 0;
    case INFO = 1;
    case WARN = 2;
    case ERROR = 3;
}

/**
 * Simple logger utility
 */
class Logger
{
    private LogLevel $level;

    /**
     * Create a new logger
     *
     * @param LogLevel $level Minimum log level to display
     */
    public function __construct(LogLevel $level = LogLevel::INFO)
    {
        $this->level = $level;
    }

    /**
     * Log debug message
     *
     * @param string $message Log message
     * @param mixed ...$args Additional arguments
     */
    public function debug(string $message, mixed ...$args): void
    {
        $this->log(LogLevel::DEBUG, $message, ...$args);
    }

    /**
     * Log info message
     *
     * @param string $message Log message
     * @param mixed ...$args Additional arguments
     */
    public function info(string $message, mixed ...$args): void
    {
        $this->log(LogLevel::INFO, $message, ...$args);
    }

    /**
     * Log warning message
     *
     * @param string $message Log message
     * @param mixed ...$args Additional arguments
     */
    public function warn(string $message, mixed ...$args): void
    {
        $this->log(LogLevel::WARN, $message, ...$args);
    }

    /**
     * Log error message
     *
     * @param string $message Log message
     * @param mixed ...$args Additional arguments
     */
    public function error(string $message, mixed ...$args): void
    {
        $this->log(LogLevel::ERROR, $message, ...$args);
    }

    /**
     * Set log level
     *
     * @param LogLevel $level New log level
     */
    public function setLevel(LogLevel $level): void
    {
        $this->level = $level;
    }

    /**
     * Get current log level
     *
     * @return LogLevel Current log level
     */
    public function getLevel(): LogLevel
    {
        return $this->level;
    }

    /**
     * Internal log method
     *
     * @param LogLevel $level Log level
     * @param string $message Log message
     * @param mixed ...$args Additional arguments
     */
    private function log(LogLevel $level, string $message, mixed ...$args): void
    {
        if ($level->value >= $this->level->value) {
            $timestamp = date('Y-m-d H:i:s');
            $levelName = $level->name;
            $formattedMessage = "[{$timestamp}] [{$levelName}] {$message}";

            if (!empty($args)) {
                $formattedMessage .= ' ' . json_encode($args);
            }

            echo $formattedMessage . "\n";
        }
    }
}

File: test-projects/java-sample/src/main/java/com/codegraph/sample/Logger.java
package com.codegraph.sample;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

/**
 * Simple logger utility
 */
public class Logger {
    /**
     * Log level enumeration
     */
    public enum LogLevel {
        DEBUG(0),
        INFO(1),
        WARN(2),
        ERROR(3);

        private final int value;

        LogLevel(int value) {
            this.value = value;
        }

        public int getValue() {
            return value;
        }
    }

    private LogLevel level;
    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");

    /**
     * Create a new logger
     *
     * @param level Minimum log level to display
     */
    public Logger(LogLevel level) {
        this.level = level;
    }

    /**
     * Log debug message
     *
     * @param message Log message
     */
    public void debug(String message) {
        log(LogLevel.DEBUG, message);
    }

    /**
     * Log info message
     *
     * @param message Log message
     */
    public void info(String message) {
        log(LogLevel.INFO, message);
    }

    /**
     * Log warning message
     *
     * @param message Log message
     */
    public void warn(String message) {
        log(LogLevel.WARN, message);
    }

    /**
     * Log error message
     *
     * @param message Log message
     */
    public void error(String message) {
        log(LogLevel.ERROR, message);
    }

    /**
     * Set log level
     *
     * @param level New log level
     */
    public void setLevel(LogLevel level) {
        this.level = level;
    }

    /**
     * Get current log level
     *
     * @return Current log level
     */
    public LogLevel getLevel() {
        return level;
    }

    /**
     * Internal log method
     *
     * @param level Log level
     * @param message Log message
     */
    private void log(LogLevel level, String message) {
        if (level.getValue() >= this.level.getValue()) {
            String timestamp = LocalDateTime.now().format(formatter);
            String levelName = level.name();
            String formattedMessage = String.format("[%s] [%s] %s", timestamp, levelName, message);
            System.out.println(formattedMessage);
        }
    }
}
